WooCommerce/Classes/ViewModels/CardPresentPayments/CardPresentModalConnectionFailedUpdateAddress.swift

```diff
@@ -26,13 +37,56 @@ final class CardPresentModalConnectingFailedUpdateAddress: CardPresentPaymentsMo
 
     let bottomSubtitle: String? = nil
 
-    init(continueSearch: @escaping () -> Void, cancelSearch: @escaping () -> Void) {
-        self.continueSearchAction = continueSearch
+    init(adminUrl: URL?,
+         site: Site?,
+         openUrlInSafari: @escaping (URL) -> Void,
+         retrySearch: @escaping () -> Void,
+         cancelSearch: @escaping () -> Void) {
+        self.adminUrl = adminUrl
+        self.site = site
+        self.openUrlInSafariAction = openUrlInSafari
+        self.retrySearchAction = retrySearch
         self.cancelSearchAction = cancelSearch
     }
 
     func didTapPrimaryButton(in viewController: UIViewController?) {
-        continueSearchAction()
+        guard let adminUrl = adminUrl,
+              let viewController = viewController else {
+            return retrySearchAction()
+        }
+        switch site?.isWordPressStore {
+        case true:
+            presentAuthenticatedWebview(url: adminUrl, from: viewController)
+        default:
+            self.openUrlInSafariAction(adminUrl)
+        }
+    }
+
+    private func presentAuthenticatedWebview(url adminUrl: URL, from viewController: UIViewController) {
```

Comment:
Suggestion: Should the modal be doing this (and aware of AuthenticatedWebView), or could we keep the modal more generic by having the caller deal with this?